src/js/src/views/SolrWayback.vue

```diff
@@ -41,55 +43,61 @@ export default {
+      resetState:'resetState',
+      addToSearchAppliedFacets:'addToSearchAppliedFacets',
+      emptySearchAppliedFacets:'emptySearchAppliedFacets',
       updateSolrSettingGrouping:'updateSolrSettingGrouping',
       updateSolrSettingImgSearch:'updateSolrSettingImgSearch',
       updateSolrSettingUrlSearch:'updateSolrSettingUrlSearch',
-      updateFutureSolrSettingGrouping:'updateFutureSolrSettingGrouping',
-      updateFutureSolrSettingImgSearch:'updateFutureSolrSettingImgSearch',
-      updateFutureSolrSettingUrlSearch:'updateFutureSolrSettingUrlSearch'
+      updateSolrSettingOffset:'updateSolrSettingOffset',
     }),
     onScroll(e) {
     e.target.documentElement.scrollTop > 0 ? this.scrolledFromTop = true : this.scrolledFromTop = false
     },
     backToTop() {
       window.scrollTo({ top: 0, behavior: 'smooth' })
-    }, 
+    },
   },
   beforeRouteUpdate (to, from, next) {
-    //console.log('route changed!',to)
-    this.updateQuery(to.query.q)
-    to.query.grouping === 'true' ? (this.updateSolrSettingGrouping(true), this.updateFiutureSolrSettingGrouping(true)) : (this.updateSolrSettingGrouping(false), this.updateFutureSolrSettingGrouping(false))
-    to.query.imgSearch === 'true' ? (this.updateSolrSettingImgSearch(true), this.updateFutureSolrSettingImgSearch(true)) : (this.updateSolrSettingImgSearch(false), this.updateFutureSolrSettingImgSearch(false))
-    to.query.urlSearch === 'true' ? (this.updateSolrSettingUrlSearch(true), this.updateFutureSolrSettingUrlSearch(true)) :  (this.updateSolrSettingUrlSearch(false), this.updateFutureSolrSettingUrlSearch(false))
-    to.query.facets ? this.updateSearchAppliedFacets(to.query.facets) : this.updateSearchAppliedFacets('')
-    if(this.solrSettings.imgSearch) {
-      this.requestImageSearch({query:this.query})
+    //console.log('route changed!',to.query, from.query)
+    // Check if any of our params have changed. Could be refactored into a nice functon.
+    if(to.query.query !== undefined &&
+       (to.query.query !== from.query.query ||
+       to.query.offset !== from.query.offset ||
+       to.query.imgSearch !== from.query.imgSerach ||
+       to.query.urlSearch !== from.query.urlSearch ||
+       to.query.grouping !== from.query.grouping ||
+       to.query.facets !== from.query.facets)) {
+      //console.log('we doing a route search')
+      // update our variables from the query.
+      to.query.grouping === 'true' || to.query.grouping === true ? this.updateSolrSettingGrouping(true) : this.updateSolrSettingGrouping(false)
+      to.query.imgSearch === 'true' || to.query.imgSearch === true ? this.updateSolrSettingImgSearch(true) : this.updateSolrSettingImgSearch(false)
+      to.query.urlSearch === 'true' || to.query.urlSearch === true ? this.updateSolrSettingUrlSearch(true) : this.updateSolrSettingUrlSearch(false)
+      to.query.urlSearch === 'true' || to.query.urlSearch === true ? this.updateSolrSettingUrlSearch(true) : this.updateSolrSettingUrlSearch(false)
```

Comment:
Curious - is both the check for String _true_ and Boolean _true_ needed?

Reply:
It's needed. When I was working on it, i noticed that when the backbutton is pushed, the values are no longer booleans, but strings. This is from a normal query: 

facets: ""
grouping: false
imgSearch: false
offset: 0
query: "toke"
urlSearch: false

where this is from when the backbutton has been used: 

facets: ""
grouping: "false"
imgSearch: "false"
offset: "0"
query: "hans"
urlSearch: "false"